torch/ao/quantization/pt2e/numeric_debugger.py

```diff
@@ -11,3 +20,188 @@ def generate_numeric_debug_handle(graph_module: GraphModule) -> None:
         if node.op != "placeholder" and NUMERIC_DEBUG_HANDLE_KEY not in node.meta:
             node.meta[NUMERIC_DEBUG_HANDLE_KEY] = unique_id
             unique_id += 1
+
+
+class OutputLogger(torch.nn.Module):
+    """
+    Base class for capturing output values.
+    """
+
+    # Mark as impure so that calls to it will not be removed during DCE.
+    _is_impure = True
+
+    def __init__(
+        self,
+        node_name: str,
+        nn_module_stack: object,
+        debug_handle: str,
+    ) -> None:
+        super().__init__()
+        self.node_name = node_name
+        self.nn_module_stack = nn_module_stack
+        self.debug_handle = debug_handle
+        self.stats: List[torch.Tensor] = []
+
+    def forward(self, x: object) -> object:
+        if isinstance(x, torch.Tensor):
+            self.stats.append(x.detach())
+        return x
+
+    def __repr__(self) -> str:
+        clean_dict = {
+            k: v
+            for k, v in self.__dict__.items()
+            # skip torch.nn.Module keys
+            if (k != "training") and not k.startswith("_")
+        }
+        return f"OutputLogger({clean_dict})"
```

Comment:
Why include all this stuff?
Also, I think typically subclasses of Module use "extra_repr" instead
```suggestion
    def __repr__(self) -> str:
        return f"OutputLogger(node_name={self.node_name}, nn_module_stack={self.nn_module_stack}, debug_handle={self.debug_handle}, num_stats={len(self.stats)})"
```